### PR TITLE
Личный проект: пока все дома

### PR DESCRIPTION
**Pull request by a201150209**
---
:mortar_board: [Личный проект: пока все дома](https://up.htmlacademy.ru/javascript/17/user/83947/tasks/5)

:boom: https://htmlacademy-javascript.github.io/83947-keksobooking-17/2/

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  return Math.round(min - 0.5 + Math.random() * (max - min + 1));
+}
+
+function getNumberOfImage(number) {
+  var imageNumber = number + 1;
+  imageNumber = '0' + String(imageNumber);
+  return imageNumber;
+}
+
+function PinProperty(number) {
```
Comment (rkhaslarov) [inline js/main.js]: PinEntity is more convenient here, but it's cool that you're using constructor =)

File: js/main.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+function getRandomNumberFromRange(min, max) {
+  return Math.round(min - 0.5 + Math.random() * (max - min + 1));
+}
+
+function getNumberOfImage(number) {
+  var imageNumber = number + 1;
+  imageNumber = '0' + String(imageNumber);
+  return imageNumber;
```
Comment (rkhaslarov) [inline js/main.js]: three lines of the code and the whole function, just for concating number and string?

 'img/avatars/user0' + number.toString() + '.png'

File: js/main.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      min: 130,
+      max: 630
+    }
+  }
+};
+
+function getRandomElementInArray(array) {
+  var min = 0;
+  var max = array.length - 1;
+  return array[getRandomNumberFromRange(min, max)];
```
Comment (rkhaslarov) [inline js/main.js]: unnecessary variables, they are not reused anywhere, just put the values directly =)

File: js/main.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+var NUMBER_OF_RELATED_PINS = 8;
+var map = document.querySelector('.map');
+var mapPins = document.querySelector('.map__pins');
+var mapPinsWidth = mapPins.clientWidth;
+var mapPinTemplate = document.querySelector('#pin').content.querySelector('.map__pin');
+var relatedPinsProperty = [];
+var relatedPins = [];
+var relatedPinsData = {
```
Comment (rkhaslarov) [inline js/main.js]: it's a mock data, probably would be nice to name it accordingly

File: js/main.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    x: getRandomNumberFromRange(0, mapPinsWidth),
+    y: getRandomNumberFromRange(relatedPinsData.location.y.min, relatedPinsData.location.y.max)
+  };
+}
+
+function createPinProperty(number) {
+  relatedPinsProperty.push(new PinProperty(number));
+}
+
+function createPin(property) {
```
Comment (rkhaslarov) [inline js/main.js]: renderPin is more suitable name, because for creating them you're using PinEntity

File: js/main.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  var pin = mapPinTemplate.cloneNode(true);
+  var authorAvatar = pin.querySelector('img');
+  var authorAvatarWidth = authorAvatar.width;
+  var authorAvatarHeight = authorAvatar.height;
+
+  pin.style.left = property.location.x + authorAvatarWidth / 2 + 'px';
+  pin.style.top = property.location.y - authorAvatarHeight + 'px';
+  authorAvatar.src = property.author.avatar;
+  authorAvatar.alt = '{{заголовок объявления}}'; // do not forget get alt later
+  relatedPins.push(pin);
```
Comment (rkhaslarov) [inline js/main.js]: no need to cache dom elements, just return it and use in renderPins function

File: js/main.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+function createPins() {
```
Comment (rkhaslarov) [inline js/main.js]: renderPins

File: js/main.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  var fragment = document.createDocumentFragment();
+  for (var i = 0; i < NUMBER_OF_RELATED_PINS; i++) {
+    createPinProperty(i);
+    createPin(relatedPinsProperty[i]);
```
Comment (rkhaslarov) [inline js/main.js]: fragment.appendChild(renderPin(new PinProperty(number)));

File: js/main.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    type: getRandomElementInArray(relatedPinsData.offer.type)
+  };
+  this.location = {
+    x: getRandomNumberFromRange(0, mapPinsWidth),
+    y: getRandomNumberFromRange(relatedPinsData.location.y.min, relatedPinsData.location.y.max)
+  };
+}
+
+function createPinProperty(number) {
+  relatedPinsProperty.push(new PinProperty(number));
```
Comment (rkhaslarov) [inline js/main.js]: strange function

File: js/main.js (hunk 6 above)
Comment (rkhaslarov) [reply js/main.js]: use return keyword for that

File: js/main.js (hunk 9 above)
Comment (a201150209) [reply js/main.js]: All issues were fixed, master)

**Pull request merged**